packages/dds/tree/src/shared-tree-core/transactionStack.ts

```diff
@@ -10,7 +10,11 @@ import { fail } from "../util";
  * A helper class that organizes the state needed for managing nesting transactions.
  */
 export class TransactionStack {
-	private readonly stack: { startRevision: RevisionTag; repairStore?: RepairDataStore }[] = [];
+	private readonly stack: {
+		startRevision: RevisionTag;
+		revision: RevisionTag;
```

Comment:
I'm having trouble grasping what this is and why we need it in here. It would be good to declare an interface for the whole object type so you can document the field there, then the push and pop methods can take and return an instance of that interface.

Reply:
Nice catch, this is outdated code actually. I was using it so that we can specify what revision to use for the final commit which wouldn't have been needed for every transaction anyways but now we really don't need this since we don't need to keep track of repair data per change in a commit. 😄